### ISSUE TITLE
Docs pages missing necessary landmark regions and incorrectly marked up

### ISSUE THREAD

**Issue by amberhinds** [LibreChat-AI/librechat.ai#518]
On the docs website, several expected landmark regions required for accessibility are missing.

For example, on the [Anthropic Vertex AI Object Structure](https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/anthropic_vertex) page, the only landmark regions are a main and complementary region. 
<img width="1546" height="1578" alt="Complementry region nested in main and containing the header content" src="https://github.com/user-attachments/assets/3fafce87-c21d-43d3-ab83-fdd2070ee413" />

The complementary region is actually the functional header/banner on the page with the navigation. It would be really difficult for screen reader users to find the navigation or understand the content on the page.

Recommended landmark fixes:
1. Change the current `<aside>` to a `<header>`.
2. Add a `<nav aria-label="primary">` landmark around the site navigation toggle button.
3. Add a `<nav aria-label="docs">` landmark around the doc navigation going down the left site.
4. Add a `<nav aria-label="breadcrumbs">` landmark around the breadcrumbs.
5. Add a `<nav aria-label="table of contents">` landmark around the "On this page" table of contents.
6. Add a `<nav aria-label="pagination">` landmark around the pagination links at the bottom of the article.
